src/core/os_windows.c

```diff
@@ -352,7 +385,23 @@ os_posix_fallocate(int fd, os_off_t offset, os_off_t len)
 int
 os_ftruncate(int fd, os_off_t length)
 {
-	return _chsize_s(fd, length);
+	LARGE_INTEGER distanceToMove = {0};
+	distanceToMove.QuadPart = length;
+	HANDLE handle = (HANDLE)_get_osfhandle(fd);
+	if (handle == INVALID_HANDLE_VALUE) {
+		return -1;
+	}
+	if (!SetFilePointerEx(handle, distanceToMove, NULL, FILE_BEGIN)) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	if (!SetEndOfFile(handle)) {
+		int ret = GetLastError();
+		errno = EINVAL;
+		return ret;
```

Comment:
`Upon successful completion, ftruncate() shall return 0; otherwise, −1 shall be returned and errno set to indicate the error.`

You probably want to map the error to those listed in `ftruncate` man.

Reply:
Well this is difficult to map windows error to Linux one as windows don't documents possible errors returned by those functions